arangod/MMFiles/MMFilesEngine.h

```diff
@@ -199,6 +205,18 @@ class MMFilesEngine final : public StorageEngine {
 
   Result persistLocalDocumentIds(TRI_vocbase_t& vocbase);
 
+  /// @brief regiter a recovery helper
+  /// @note not thread-safe on the assumption of static factory registration
+  static arangodb::Result registerRecoveryHelper(
+    MMFilesRecoveryHelper const& helper
+  );
+
+  /// @brief invoke visitor with each registered recovery helper
+  /// @return all recovery registered helpers invoked and returned success
+  static bool visitRecoveryHelpers(
```

Comment:
why is there no comment on thread-safety?

Reply:
because it's thread-safe to visit on the assumption that the list is not modified